### PR TITLE
add module omnidata hiveudf loader

### PR DESCRIPTION
**Pull request by wyy566**
add module omnidata hiveudf loader

### CODE REVIEW

**Comment by tushengxia** [comment]
/check-cla

> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> The following commits have not yet signed CLA.
> 
> **[62a3421](https://github.com/kunpengcompute/boostkit-bigdata/commit/62a3421dde0c0ff36d0d3ec0b5fc66337dc419ca)** | add module omnidata hiveudf loader
> 
> memo **Please access [here](https://clasign.osinfra.cn/sign/Z2l0aHViJTJGa3VucGVuZ2NvbXB1dGU=) to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment: **/check-cla** to verify. Thanks.
> 
> * If you've already signed a CLA, it's possible you're using a different email address for your gitee account. Check your existing CLA data and verify the email.
> * If you signed the CLA as an employee or a member of an organization, please contact your corporation or organization to verify you have been activated to start contributing.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please feel free to file an issue.

/check-cla

**Review by tedliUCAS — APPROVED** [review]

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/type/PrestoTypes.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        checkArgument(DECIMAL.equals(type.getBase()) && type.getParameters().size() == 2,
+                "Invalid decimal type " + type);
+        int precision = type.getParameters().get(0).getLongLiteral().intValue();
+        int scale = type.getParameters().get(1).getLongLiteral().intValue();
+        return DecimalType.createDecimalType(precision, scale);
+    }
+
+    public static Type fromObjectInspector(ObjectInspector inspector, TypeManager typeManager)
+    {
+        switch (inspector.getCategory()) {
```
Comment (tushengxia) [inline omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/type/PrestoTypes.java:71]: inspector是否存在空的场景

Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/type/PrestoTypes.java:74]: 这里的入参是hive UDF初始化返回值，返回值用于限定UDF方法对应的返回类型，不确定是否存在空场景，已补充空场景的处理。

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionNamespaceManagerFactory.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        }
+        catch (PrestoException e) {
+            throw e;
+        }
+    }
+
+    private ClassLoader getClassLoader(String catalogName)
+    {
+        String configFile = System.getProperty("user.dir") + File.separatorChar + FUNCTION_PROPERTIES_FILE_PATH + File.separatorChar + catalogName + ".properties";
+        File file = new File(configFile);
```
[the remark this answers is not in the document]
Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionNamespaceManagerFactory.java:115]: 代码117行已经对文件进行校验，判断是否存在该配置文件。

[the remark this answers is not in the document]
Comment (tushengxia) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionNamespaceManagerFactory.java:115]: 可以参考下编程规范

[the remark this answers is not in the document]
Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionNamespaceManagerFactory.java:120]: 已修改

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/FunctionRegistry.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    public static void addFunction(String functionName, Class<?> cls)
+    {
+        if (GenericUDF.class.isAssignableFrom(cls)) {
+            system.registerGenericUDF(functionName, (Class<? extends GenericUDF>) cls);
+        }
+        else if (UDF.class.isAssignableFrom(cls)) {
+            system.registerUDF(functionName, (Class<? extends UDF>) cls, false);
+        }
+        else {
+            log.warn("Function %s is not extend from GenericUDF or UDF, please check.", functionName);
```
Comment (luodan000) [inline omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/FunctionRegistry.java:54]: 应该是 is not extended

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    public static PrestoException unsupportedType(Type type)
+    {
+        return new PrestoException(HIVE_FUNCTION_UNSUPPORTED_PRESTO_TYPE, "Unsupported Presto type " + type);
+    }
+
+    public static PrestoException unsupportedType(TypeSignature type)
+    {
+        return new PrestoException(HIVE_FUNCTION_UNSUPPORTED_PRESTO_TYPE, "Unsupported Presto type " + type);
```
Comment (luodan000) [inline omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java:65]: 次行的日志和60行的一模一样，查看日志时不易区分，建议区分开来

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    public static PrestoException unsupportedFunctionType(Class<?> cls, Throwable t)
+    {
+        return new PrestoException(HIVE_FUNCTION_UNSUPPORTED_FUNCTION_TYPE,
+                format("Unsupported function type %s / %s", cls.getName(), cls.getSuperclass().getName()), t);
+    }
+
+    public static PrestoException unsupportedNamespace(QualifiedObjectName name)
+    {
+        return new PrestoException(HIVE_FUNCTION_UNSUPPORTED_NAMESPACE, "Hive udf unsupported namespace " + name.getCatalogSchemaName()
+                + ". It's schema should be default.");
```
Comment (luodan000) [inline omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java:88]: 应该是 Its schema

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    public static PrestoException initializationError(Throwable t)
+    {
+        return new PrestoException(HIVE_FUNCTION_INITIALIZATION_ERROR, t);
+    }
+
+    public static PrestoException initializationError(String filePath, Exception e)
+    {
+        return new PrestoException(HIVE_FUNCTION_INITIALIZATION_ERROR, format("Read configuration %s fail. %s", filePath, e.getMessage()));
```
Comment (luodan000) [inline omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java:103]: 使用正确的语法

Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java]: 已修改

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java (hunk 5 above)
Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java]: 已修改

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java (hunk 4 above)
Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/HiveFunctionErrorCode.java]: 已修改

File: omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/FunctionRegistry.java (hunk 3 above)
Comment (wyy566) [reply omnidata/omnidata-hiveudf-loader/src/main/java/io/prestosql/plugin/hive/functions/FunctionRegistry.java]: 已修改

**Pull request merged**